app/src/test/java/com/edulectronics/tinycircuit/LightbulbTest.java

```diff
@@ -26,13 +26,29 @@ public void connectedLightBulbIsOn(){
         Output bulbOutput = bulb.getOutputByIndex(0);
         Input bulbInput = bulb.getInputByIndex(0);
 
-        Connector.getInstance().connect(bulbInput, powerOutput);
-        Connector.getInstance().connect(powerInput, bulbOutput);
+        Connector.connect(bulbInput, powerOutput);
+        Connector.connect(powerInput, bulbOutput);
 
         powersource.startConnection();
         assertTrue(bulb.isOn);
     }
 
+    @Test
+    public void weakPowersourceLightbulbNotOn(){
+        Powersource powersource = new Powersource(3);
+        Output powerOutput = powersource.getOutputByIndex(0);
+        Input powerInput = powersource.getInputByIndex(0);
+
+        Lightbulb bulb = new Lightbulb();
+        Output bulbOutput = bulb.getOutputByIndex(0);
+        Input bulbInput = bulb.getInputByIndex(0);
```

Comment:
Om de code wat leesbaarder te maken zou ik 4 public constants in de algemene component class zetten LEFT = 0, TOP = 1, etc. Dan kun je de methode getInputBySide(int side) noemen. Dus bijvoorbeeld getInputBySide(LEFT).